packages/debug/src/browser/model/debug-thread.tsx

```diff
@@ -61,6 +61,14 @@ export class DebugThread extends DebugThreadData implements TreeElement {
         this.onDidChangedEmitter.fire(undefined);
     }
 
+    protected _framesDirty: boolean = false;
```

Comment:
I don't like this design, we introduce some property which gets updated in adhoc fashion by clients and this class, never used by class. Feels like patching. Please have a look what VS Code does on stop and why it works for them. If we should clear frames on stop then something should happen in `update` method.

Reply:
Relevant code in VS Code: https://github.com/microsoft/vscode/blob/c520c0c69daf406540aa1645773e5f6afed81427/src/vs/workbench/contrib/debug/browser/debugSession.ts#L619 You should be able to debug it by using Electron devtools or just read carefully.

Reply:
Maybe to put it another way: there is not use in keeping stack frames that we don't want to use anymore. The additional info that we used to have stack frames, but now they are invalid is not being used in our UI, so we might just use the lack of stack frames as an indicator that we need to fetch them.

Reply:
I didn't see that we had such a map of frames before you mentioned it. So we make `clearFrames` public and use that when we handle a stop event?

Reply:
Alright, apparently we already had a `clearFrames` call every time a thread stopped. The patch is now changed to detect that and is now really minimal :)